### PR TITLE
Flag SQL Injection attacks at TransferName

### PR DESCRIPTION
**Pull request by pjol**
Added a regexp check in the TransferName function to verify that contract addresses are correctly formatted and not malicious.

### CODE REVIEW

File: internal/events/handlers.go [hunk 1]
```diff
@@ -31,7 +31,12 @@ func (s *Service) AddEvent(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
 	// check whether event already exists
-	name := s.db.TransferName(ev.Contract)
+	name, flag := s.db.TransferName(ev.Contract)
+	if flag {
+		w.WriteHeader(http.StatusForbidden)
```
Comment (kev-techi) [inline internal/events/handlers.go:36]: for a request response complaining about badly formatted parameters, I would return a 400 instead. Bad request makes more sense to me.

File: internal/services/db/db.go [hunk 2]
```diff
@@ -282,13 +289,23 @@ func (db *DB) PushTokenTableExists(suffix string) (bool, error) {
 }
 
 // TransferName returns the name of the transfer db for the given contract
-func (d *DB) TransferName(contract string) string {
-	return fmt.Sprintf("%v_%s", d.chainID, strings.ToLower(contract))
+func (d *DB) TransferName(contract string) (string, bool) {
```
Comment (kev-techi) [inline internal/services/db/db.go:292]: This should probably just return `errors.New("Bad contract address")`. And then we can do the standard:

```
name, err := d.TransferName(ev.Contract)
		if err  != nil {
			return err
		}
```

Comment (kev-techi) [reply internal/services/db/db.go:292]: I would also suggest renaming this method to something that matches what it does (my fault for giving it a confusing name).

How about `TableNameSuffix`?

**Review by kev-techi — CHANGES_REQUESTED** [review]
All in all, looks good. Just a few comments.

**Comment by pjol** [comment]
Continued at PR #38 Flagging SQL Injection at TableNameSuffix

**Pull request closed without merging**